src/app/clusters/thermostat-server/PresetStructWithOwnedMembers.cpp

```diff
@@ -0,0 +1,155 @@
+/**
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "PresetStructWithOwnedMembers.h"
+
+using namespace chip;
+using namespace chip::app;
+using namespace DataModel;
+using namespace chip::app::Clusters::Thermostat::Structs;
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace Thermostat {
+
+PresetStructWithOwnedMembers::PresetStructWithOwnedMembers(const PresetStruct::Type & other)
+{
+    *this = other;
+}
+
+void PresetStructWithOwnedMembers::operator=(const PresetStruct::Type & other)
+{
+    SetPresetScenario(other.presetScenario);
+    CHIP_ERROR err = SetPresetHandle(other.presetHandle);
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(Zcl, "Failed to set Preset handle with err %" CHIP_ERROR_FORMAT, err.Format());
+    }
+    err = SetName(other.name);
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(Zcl, "Failed to set Preset name with err %" CHIP_ERROR_FORMAT, err.Format());
+    }
+    SetCoolingSetpoint(other.coolingSetpoint);
+    SetHeatingSetpoint(other.heatingSetpoint);
+    SetBuiltIn(other.builtIn);
+}
+
+void PresetStructWithOwnedMembers::SetPresetScenario(PresetScenarioEnum enumValue)
+{
+    presetScenario = enumValue;
+}
+
+CHIP_ERROR PresetStructWithOwnedMembers::SetPresetHandle(const Nullable<ByteSpan> & newPresetHandle)
+{
+    if (!newPresetHandle.IsNull())
+    {
+        size_t newPresetHandleSize = newPresetHandle.Value().size();
+        if (newPresetHandleSize > kPresetHandleSize)
+        {
+            ChipLogError(Zcl, "Failed to set Preset handle. New preset handle size (%u) > allowed preset handle size (%u)",
+                         static_cast<uint8_t>(newPresetHandleSize), static_cast<uint8_t>(kPresetNameSize));
```

Comment:
This was in the previous review....
```suggestion
                         static_cast<unsigned>(newPresetHandleSize), static_cast<unsigned>(kPresetNameSize));
```

Reply:
i missed this. sorry